xl-v1-multipart.go

```diff
@@ -404,6 +405,32 @@ func (xl xlObjects) CompleteMultipartUpload(bucket string, object string, upload
 		return "", toObjectErr(err, bucket, object)
 	}
 
+	// Remove parts that weren't present in CompleteMultipartUpload request
+	for _, curpart := range currentXLMeta.Parts {
```

Comment:
Move this into a function in `xl-v1-multipart-common.go` that way all you have to take is 

```bucket, object, uploadID, curpart.Name``` arguments.